TestFatalThrowBreakpoint systematically fails on Go 1.17

Is the version test for 1.16 supposed to be extended to 1.17? cc @suzmue

```

DAP server listening at: [::]:36999
2021-07-20T09:13:18+02:00 debug layer=dap DAP server pid = 138195
Connecting to server at: [::]:36999
2021-07-20T09:13:18+02:00 debug layer=dap [<- from client]{"seq":1,"type":"request","command":"initialize","arguments":{"adapterID":"go","locale":"en-us","linesStartAt1":true,"columnsStartAt1":true,"pathFormat":"path","supportsVariableType":true,"supportsVariablePaging":true,"supportsRunInTerminalRequest":true}}
2021-07-20T09:13:18+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":1,"success":true,"command":"initialize","body":{"supportsConfigurationDoneRequest":true,"supportsFunctionBreakpoints":true,"supportsConditionalBreakpoints":true,"supportsEvaluateForHovers":true,"supportsSetVariable":true,"supportsExceptionInfoRequest":true,"supportTerminateDebuggee":true,"supportsDelayedStackTraceLoading":true,"supportsClipboardContext":true}}
2021-07-20T09:13:18+02:00 debug layer=dap [<- from client]{"seq":2,"type":"request","command":"launch","arguments":{"mode":"exec","program":"/tmp/fatalerror.6f73fa34","request":"launch","stopOnEntry":false}}
2021-07-20T09:13:18+02:00 debug layer=dap running program in /tmp

2021-07-20T09:13:18+02:00 debug layer=dap [-> to client]{"seq":0,"type":"event","event":"initialized"}
2021-07-20T09:13:18+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":2,"success":true,"command":"launch"}
2021-07-20T09:13:18+02:00 debug layer=dap [<- from client]{"seq":3,"type":"request","command":"setBreakpoints","arguments":{"source":{"name":"fatalerror.go","path":"/home/aa/n/delve/_fixtures/fatalerror.go"},"breakpoints":[{"line":3}]}}
2021-07-20T09:13:18+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":3,"success":true,"command":"setBreakpoints","body":{"breakpoints":[{"id":1,"verified":true,"source":{"name":"fatalerror.go","path":"/home/aa/n/delve/_fixtures/fatalerror.go"},"line":3}]}}
2021-07-20T09:13:18+02:00 debug layer=dap [<- from client]{"seq":4,"type":"request","command":"configurationDone","arguments":{}}
2021-07-20T09:13:18+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":4,"success":true,"command":"configurationDone"}
2021-07-20T09:13:18+02:00 debug layer=dap "continue" command stopped - reason "breakpoint", location /home/aa/n/delve/_fixtures/fatalerror.go:3
2021-07-20T09:13:18+02:00 debug layer=dap [-> to client]{"seq":0,"type":"event","event":"stopped","body":{"reason":"breakpoint","threadId":1,"allThreadsStopped":true,"hitBreakpointIds":[1]}}
2021-07-20T09:13:18+02:00 debug layer=dap [<- from client]{"seq":5,"type":"request","command":"threads"}
2021-07-20T09:13:18+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":5,"success":true,"command":"threads","body":{"threads":[{"id":1,"name":"* [Go 1] main.main (Thread 138262)"},{"id":2,"name":"[Go 2] runtime.gopark"},{"id":3,"name":"[Go 3] runtime.gopark"},{"id":4,"name":"[Go 4] runtime.gopark"}]}}
2021-07-20T09:13:18+02:00 debug layer=dap [<- from client]{"seq":6,"type":"request","command":"stackTrace","arguments":{"threadId":1,"levels":20,"format":{}}}
2021-07-20T09:13:18+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":6,"success":true,"command":"stackTrace","body":{"stackFrames":[{"id":1000,"name":"main.main","source":{"name":"fatalerror.go","path":"/home/aa/n/delve/_fixtures/fatalerror.go"},"line":3,"column":0},{"id":1001,"name":"runtime.main","source":{"name":"proc.go","path":"/usr/local/go-tip/src/runtime/proc.go","presentationHint":"deemphasize"},"line":255,"column":0},{"id":1002,"name":"runtime.goexit","source":{"name":"asm_amd64.s","path":"/usr/local/go-tip/src/runtime/asm_amd64.s","presentationHint":"deemphasize"},"line":1581,"column":0}],"totalFrames":3}}
2021-07-20T09:13:18+02:00 debug layer=dap [<- from client]{"seq":7,"type":"request","command":"scopes","arguments":{"frameId":1000}}
2021-07-20T09:13:18+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":7,"success":true,"command":"scopes","body":{"scopes":[{"name":"Arguments","variablesReference":1000,"expensive":false,"source":{}},{"name":"Locals","variablesReference":1001,"expensive":false,"source":{}}]}}
2021-07-20T09:13:18+02:00 debug layer=dap [<- from client]{"seq":8,"type":"request","command":"variables","arguments":{"variablesReference":1000,"format":{}}}
2021-07-20T09:13:18+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":8,"success":true,"command":"variables","body":{"variables":[]}}
2021-07-20T09:13:18+02:00 debug layer=dap [<- from client]{"seq":9,"type":"request","command":"variables","arguments":{"variablesReference":1001,"format":{}}}
2021-07-20T09:13:18+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":9,"success":true,"command":"variables","body":{"variables":[]}}
2021-07-20T09:13:18+02:00 debug layer=dap [<- from client]{"seq":10,"type":"request","command":"continue","arguments":{"threadId":1}}
2021-07-20T09:13:18+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":10,"success":true,"command":"continue","body":{"allThreadsContinued":true}}
fatal error: go of nil func value
2021-07-20T09:13:18+02:00 debug layer=dap "continue" command stopped - reason "breakpoint", location /usr/local/go-tip/src/runtime/panic.go:1244
2021-07-20T09:13:18+02:00 debug layer=dap [-> to client]{"seq":0,"type":"event","event":"stopped","body":{"reason":"exception","description":"fatal error","threadId":1,"allThreadsStopped":true,"hitBreakpointIds":[-2]}}
    server_test.go:3713: 
        got  &dap.StoppedEvent{Event:dap.Event{ProtocolMessage:dap.ProtocolMessage{Seq:0, Type:"event"}, Event:"stopped"}, Body:dap.StoppedEventBody{Reason:"exception", Description:"fatal error", ThreadId:1, PreserveFocusHint:false, Text:"", AllThreadsStopped:true, HitBreakpointIds:[]int{-2}}}
        want ThreadId=1 Reason="exception" Description="fatal error" Text="\"go of nil func value\""
2021-07-20T09:13:18+02:00 debug layer=dap [<- from client]{"seq":11,"type":"request","command":"exceptionInfo","arguments":{"threadId":1}}
2021-07-20T09:13:18+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":11,"success":true,"command":"exceptionInfo","body":{"exceptionId":"fatal error","description":"Error getting throw reason: could not find loclist entry at 0x2b511 for address 0x42e5f1","breakMode":"","details":{"stackTrace":"Stack:\n\t0  0x000000000042e7e0 in runtime.fatalthrow\n\t    at /usr/local/go-tip/src/runtime/panic.go:1244\n\t1  0x000000000042e5f1 in runtime.throw\n\t    at /usr/local/go-tip/src/runtime/panic.go:1198\n\t2  0x00000000004389ae in runtime.newproc1\n\t    at /usr/local/go-tip/src/runtime/proc.go:4288\n\t3  0x000000000043864f in runtime.newproc.func1\n\t    at /usr/local/go-tip/src/runtime/proc.go:4255\n\t4  0x0000000000456869 in runtime.systemstack\n\t    at /usr/local/go-tip/src/runtime/asm_amd64.s:383\n\t5  0x0000000000458fa9 in runtime.newproc\n\t    at \u003cautogenerated\u003e:1\n"}}}
    server_test.go:3724: 
        got  &dap.ExceptionInfoResponse{Response:dap.Response{ProtocolMessage:dap.ProtocolMessage{Seq:0, Type:"response"}, RequestSeq:11, Success:true, Command:"exceptionInfo", Message:""}, Body:dap.ExceptionInfoResponseBody{ExceptionId:"fatal error", Description:"Error getting throw reason: could not find loclist entry at 0x2b511 for address 0x42e5f1", BreakMode:"", Details:dap.ExceptionDetails{Message:"", TypeName:"", FullTypeName:"", EvaluateName:"", StackTrace:"Stack:\n\t0  0x000000000042e7e0 in runtime.fatalthrow\n\t    at /usr/local/go-tip/src/runtime/panic.go:1244\n\t1  0x000000000042e5f1 in runtime.throw\n\t    at /usr/local/go-tip/src/runtime/panic.go:1198\n\t2  0x00000000004389ae in runtime.newproc1\n\t    at /usr/local/go-tip/src/runtime/proc.go:4288\n\t3  0x000000000043864f in runtime.newproc.func1\n\t    at /usr/local/go-tip/src/runtime/proc.go:4255\n\t4  0x0000000000456869 in runtime.systemstack\n\t    at /usr/local/go-tip/src/runtime/asm_amd64.s:383\n\t5  0x0000000000458fa9 in runtime.newproc\n\t    at <autogenerated>:1\n", InnerException:[]dap.ExceptionDetails(nil)}}}
        want ExceptionId="runtime error" Text="go of nil func value"
2021-07-20T09:13:18+02:00 debug layer=dap [<- from client]{"seq":12,"type":"request","command":"disconnect","arguments":{"terminateDebuggee":true}}
2021-07-20T09:13:18+02:00 debug layer=dap [-> to client]{"seq":0,"type":"event","event":"output","body":{"category":"console","output":"Detaching and terminating target process\n","source":{}}}
2021-07-20T09:13:18+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":12,"success":true,"command":"disconnect"}
2021-07-20T09:13:18+02:00 debug layer=dap [-> to client]{"seq":0,"type":"event","event":"terminated","body":{}}
2021-07-20T09:13:18+02:00 debug layer=dap DAP server stopping...
2021-07-20T09:13:18+02:00 debug layer=dap DAP server stopped
DAP server listening at: [::]:42069
2021-07-20T09:13:19+02:00 debug layer=dap DAP server pid = 138195
Connecting to server at: [::]:42069
2021-07-20T09:13:19+02:00 debug layer=dap [<- from client]{"seq":1,"type":"request","command":"initialize","arguments":{"adapterID":"go","locale":"en-us","linesStartAt1":true,"columnsStartAt1":true,"pathFormat":"path","supportsVariableType":true,"supportsVariablePaging":true,"supportsRunInTerminalRequest":true}}
2021-07-20T09:13:19+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":1,"success":true,"command":"initialize","body":{"supportsConfigurationDoneRequest":true,"supportsFunctionBreakpoints":true,"supportsConditionalBreakpoints":true,"supportsEvaluateForHovers":true,"supportsSetVariable":true,"supportsExceptionInfoRequest":true,"supportTerminateDebuggee":true,"supportsDelayedStackTraceLoading":true,"supportsClipboardContext":true}}
2021-07-20T09:13:19+02:00 debug layer=dap [<- from client]{"seq":2,"type":"request","command":"launch","arguments":{"mode":"exec","program":"/tmp/testdeadlock.0a2edf99","request":"launch","stopOnEntry":false}}
2021-07-20T09:13:19+02:00 debug layer=dap running program in /tmp

2021-07-20T09:13:19+02:00 debug layer=dap [-> to client]{"seq":0,"type":"event","event":"initialized"}
2021-07-20T09:13:19+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":2,"success":true,"command":"launch"}
2021-07-20T09:13:19+02:00 debug layer=dap [<- from client]{"seq":3,"type":"request","command":"setBreakpoints","arguments":{"source":{"name":"testdeadlock.go","path":"/home/aa/n/delve/_fixtures/testdeadlock.go"},"breakpoints":[{"line":3}]}}
2021-07-20T09:13:19+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":3,"success":true,"command":"setBreakpoints","body":{"breakpoints":[{"id":1,"verified":true,"source":{"name":"testdeadlock.go","path":"/home/aa/n/delve/_fixtures/testdeadlock.go"},"line":3}]}}
2021-07-20T09:13:19+02:00 debug layer=dap [<- from client]{"seq":4,"type":"request","command":"configurationDone","arguments":{}}
2021-07-20T09:13:19+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":4,"success":true,"command":"configurationDone"}
2021-07-20T09:13:19+02:00 debug layer=dap "continue" command stopped - reason "breakpoint", location /home/aa/n/delve/_fixtures/testdeadlock.go:3
2021-07-20T09:13:19+02:00 debug layer=dap [-> to client]{"seq":0,"type":"event","event":"stopped","body":{"reason":"breakpoint","threadId":1,"allThreadsStopped":true,"hitBreakpointIds":[1]}}
2021-07-20T09:13:19+02:00 debug layer=dap [<- from client]{"seq":5,"type":"request","command":"threads"}
2021-07-20T09:13:19+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":5,"success":true,"command":"threads","body":{"threads":[{"id":1,"name":"* [Go 1] main.main (Thread 138316)"},{"id":2,"name":"[Go 2] runtime.gopark"},{"id":3,"name":"[Go 3] runtime.gopark"},{"id":4,"name":"[Go 4] runtime.gopark"}]}}
2021-07-20T09:13:19+02:00 debug layer=dap [<- from client]{"seq":6,"type":"request","command":"stackTrace","arguments":{"threadId":1,"levels":20,"format":{}}}
2021-07-20T09:13:19+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":6,"success":true,"command":"stackTrace","body":{"stackFrames":[{"id":1000,"name":"main.main","source":{"name":"testdeadlock.go","path":"/home/aa/n/delve/_fixtures/testdeadlock.go"},"line":3,"column":0},{"id":1001,"name":"runtime.main","source":{"name":"proc.go","path":"/usr/local/go-tip/src/runtime/proc.go","presentationHint":"deemphasize"},"line":255,"column":0},{"id":1002,"name":"runtime.goexit","source":{"name":"asm_amd64.s","path":"/usr/local/go-tip/src/runtime/asm_amd64.s","presentationHint":"deemphasize"},"line":1581,"column":0}],"totalFrames":3}}
2021-07-20T09:13:19+02:00 debug layer=dap [<- from client]{"seq":7,"type":"request","command":"scopes","arguments":{"frameId":1000}}
2021-07-20T09:13:19+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":7,"success":true,"command":"scopes","body":{"scopes":[{"name":"Arguments","variablesReference":1000,"expensive":false,"source":{}},{"name":"Locals","variablesReference":1001,"expensive":false,"source":{}}]}}
2021-07-20T09:13:19+02:00 debug layer=dap [<- from client]{"seq":8,"type":"request","command":"variables","arguments":{"variablesReference":1000,"format":{}}}
2021-07-20T09:13:19+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":8,"success":true,"command":"variables","body":{"variables":[]}}
2021-07-20T09:13:19+02:00 debug layer=dap [<- from client]{"seq":9,"type":"request","command":"variables","arguments":{"variablesReference":1001,"format":{}}}
2021-07-20T09:13:19+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":9,"success":true,"command":"variables","body":{"variables":[]}}
2021-07-20T09:13:19+02:00 debug layer=dap [<- from client]{"seq":10,"type":"request","command":"continue","arguments":{"threadId":1}}
2021-07-20T09:13:19+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":10,"success":true,"command":"continue","body":{"allThreadsContinued":true}}
fatal error: all goroutines are asleep - deadlock!
2021-07-20T09:13:19+02:00 debug layer=dap "continue" command stopped - reason "breakpoint", location /usr/local/go-tip/src/runtime/panic.go:1244
2021-07-20T09:13:19+02:00 debug layer=dap [-> to client]{"seq":0,"type":"event","event":"stopped","body":{"reason":"exception","description":"fatal error","allThreadsStopped":true,"hitBreakpointIds":[-2]}}
2021-07-20T09:13:19+02:00 debug layer=dap [<- from client]{"seq":11,"type":"request","command":"disconnect","arguments":{"terminateDebuggee":true}}
2021-07-20T09:13:19+02:00 debug layer=dap [-> to client]{"seq":0,"type":"event","event":"output","body":{"category":"console","output":"Detaching and terminating target process\n","source":{}}}
2021-07-20T09:13:19+02:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":11,"success":true,"command":"disconnect"}
2021-07-20T09:13:19+02:00 debug layer=dap [-> to client]{"seq":0,"type":"event","event":"terminated","body":{}}
2021-07-20T09:13:19+02:00 debug layer=dap DAP server stopping...
2021-07-20T09:13:19+02:00 debug layer=dap DAP server stopped
--- FAIL: TestFatalThrowBreakpoint (1.04s)
FAIL
exit status 1
FAIL	github.com/go-delve/delve/service/dap	1.042s
Error executing command: go test -run TestFatalThrowBreakpoint -v -log-output=dap
```

--- a/pkg/proc/amd64_arch.go
+++ b/pkg/proc/amd64_arch.go
@@ -219,7 +219,7 @@ func amd64SwitchStack(it *stackIterator, _ *op.DwarfRegisters) bool {
 		return true
 
 	default:
-		if it.systemstack && it.top && it.g != nil && strings.HasPrefix(it.frame.Current.Fn.Name, "runtime.") && it.frame.Current.Fn.Name != "runtime.throw" {
+		if it.systemstack && it.top && it.g != nil && strings.HasPrefix(it.frame.Current.Fn.Name, "runtime.") && it.frame.Current.Fn.Name != "runtime.throw" && it.frame.Current.Fn.Name != "runtime.fatalthrow" {
 			// The runtime switches to the system stack in multiple places.
 			// This usually happens through a call to runtime.systemstack but there
 			// are functions that switch to the system stack manually (for example

--- a/pkg/proc/amd64_arch.go
+++ b/pkg/proc/amd64_arch.go
@@ -219,7 +219,7 @@ func amd64SwitchStack(it *stackIterator, _ *op.DwarfRegisters) bool {
 		return true
 
 	default:
-		if it.systemstack && it.top && it.g != nil && strings.HasPrefix(it.frame.Current.Fn.Name, "runtime.") && it.frame.Current.Fn.Name != "runtime.fatalthrow" {
+		if it.systemstack && it.top && it.g != nil && strings.HasPrefix(it.frame.Current.Fn.Name, "runtime.") && it.frame.Current.Fn.Name != "runtime.throw" {
 			// The runtime switches to the system stack in multiple places.
 			// This usually happens through a call to runtime.systemstack but there
 			// are functions that switch to the system stack manually (for example
@@ -228,7 +228,7 @@ func amd64SwitchStack(it *stackIterator, _ *op.DwarfRegisters) bool {
 			// calls we switch directly to the goroutine stack if we detect that the
 			// function at the top of the stack is a runtime function.
 			//
-			// The function "runtime.fatalthrow" is deliberately excluded from this
+			// The function "runtime.throw" is deliberately excluded from this
 			// because it can end up in the stack during a cgo call and switching to
 			// the goroutine stack will exclude all the C functions from the stack
 			// trace.

--- a/pkg/proc/arm64_arch.go
+++ b/pkg/proc/arm64_arch.go
@@ -156,7 +156,7 @@ func arm64SwitchStack(it *stackIterator, callFrameRegs *op.DwarfRegisters) bool 
 			it.pc = newlr
 			return true
 		default:
-			if it.systemstack && it.top && it.g != nil && strings.HasPrefix(it.frame.Current.Fn.Name, "runtime.") && it.frame.Current.Fn.Name != "runtime.throw" {
+			if it.systemstack && it.top && it.g != nil && strings.HasPrefix(it.frame.Current.Fn.Name, "runtime.") && it.frame.Current.Fn.Name != "runtime.throw" && it.frame.Current.Fn.Name != "runtime.fatalthrow" {
 				// The runtime switches to the system stack in multiple places.
 				// This usually happens through a call to runtime.systemstack but there
 				// are functions that switch to the system stack manually (for example

--- a/pkg/proc/arm64_arch.go
+++ b/pkg/proc/arm64_arch.go
@@ -156,7 +156,7 @@ func arm64SwitchStack(it *stackIterator, callFrameRegs *op.DwarfRegisters) bool 
 			it.pc = newlr
 			return true
 		default:
-			if it.systemstack && it.top && it.g != nil && strings.HasPrefix(it.frame.Current.Fn.Name, "runtime.") && it.frame.Current.Fn.Name != "runtime.fatalthrow" {
+			if it.systemstack && it.top && it.g != nil && strings.HasPrefix(it.frame.Current.Fn.Name, "runtime.") && it.frame.Current.Fn.Name != "runtime.throw" {
 				// The runtime switches to the system stack in multiple places.
 				// This usually happens through a call to runtime.systemstack but there
 				// are functions that switch to the system stack manually (for example

--- a/pkg/proc/i386_arch.go
+++ b/pkg/proc/i386_arch.go
@@ -148,7 +148,7 @@ func i386SwitchStack(it *stackIterator, _ *op.DwarfRegisters) bool {
 		return true
 
 	default:
-		if it.systemstack && it.top && it.g != nil && strings.HasPrefix(it.frame.Current.Fn.Name, "runtime.") && it.frame.Current.Fn.Name != "runtime.throw" {
+		if it.systemstack && it.top && it.g != nil && strings.HasPrefix(it.frame.Current.Fn.Name, "runtime.") && it.frame.Current.Fn.Name != "runtime.throw" && it.frame.Current.Fn.Name != "runtime.fatalthrow" {
 			// The runtime switches to the system stack in multiple places.
 			// This usually happens through a call to runtime.systemstack but there
 			// are functions that switch to the system stack manually (for example

--- a/pkg/proc/i386_arch.go
+++ b/pkg/proc/i386_arch.go
@@ -148,7 +148,7 @@ func i386SwitchStack(it *stackIterator, _ *op.DwarfRegisters) bool {
 		return true
 
 	default:
-		if it.systemstack && it.top && it.g != nil && strings.HasPrefix(it.frame.Current.Fn.Name, "runtime.") && it.frame.Current.Fn.Name != "runtime.fatalthrow" {
+		if it.systemstack && it.top && it.g != nil && strings.HasPrefix(it.frame.Current.Fn.Name, "runtime.") && it.frame.Current.Fn.Name != "runtime.throw" {
 			// The runtime switches to the system stack in multiple places.
 			// This usually happens through a call to runtime.systemstack but there
 			// are functions that switch to the system stack manually (for example
@@ -157,7 +157,7 @@ func i386SwitchStack(it *stackIterator, _ *op.DwarfRegisters) bool {
 			// calls we switch directly to the goroutine stack if we detect that the
 			// function at the top of the stack is a runtime function.
 			//
-			// The function "runtime.fatalthrow" is deliberately excluded from this
+			// The function "runtime.throw" is deliberately excluded from this
 			// because it can end up in the stack during a cgo call and switching to
 			// the goroutine stack will exclude all the C functions from the stack
 			// trace.

--- a/pkg/proc/proc_test.go
+++ b/pkg/proc/proc_test.go
@@ -1354,14 +1354,7 @@ func TestThreadFrameEvaluation(t *testing.T) {
 		scope, err := proc.ConvertEvalScope(p, 0, 0, 0)
 		assertNoError(err, t, "ConvertEvalScope() on frame 0")
 		_, err = scope.EvalVariable("s", normalLoadConfig)
-		if err == nil {
-			t.Errorf("expected error for EvalVariable(\"s\") on frame 0")
-		}
-
-		scope, err = proc.ConvertEvalScope(p, 0, 1, 0)
-		assertNoError(err, t, "ConvertEvalScope() on frame 1")
-		_, err = scope.EvalVariable("s", normalLoadConfig)
-		assertNoError(err, t, "EvalVariable(\"s\") on frame 1")
+		assertNoError(err, t, "EvalVariable(\"s\") on frame 0")
 	})
 }
 

--- a/pkg/proc/target.go
+++ b/pkg/proc/target.go
@@ -375,7 +375,7 @@ func (t *Target) createUnrecoveredPanicBreakpoint() {
 
 // createFatalThrowBreakpoint creates the a breakpoint as runtime.fatalthrow.
 func (t *Target) createFatalThrowBreakpoint() {
-	fatalpcs, err := FindFunctionLocation(t.Process, "runtime.fatalthrow", 0)
+	fatalpcs, err := FindFunctionLocation(t.Process, "runtime.throw", 0)
 	if err == nil {
 		bp, err := t.SetBreakpointWithID(fatalThrowID, fatalpcs[0])
 		if err == nil {

--- a/service/dap/server.go
+++ b/service/dap/server.go
@@ -2731,7 +2731,7 @@ func (s *Server) onExceptionInfoRequest(request *dap.ExceptionInfoRequest) {
 }
 
 func (s *Server) throwReason(goroutineID int) (string, error) {
-	return s.getExprString("s", goroutineID, 1)
+	return s.getExprString("s", goroutineID, 0)
 }
 
 func (s *Server) panicReason(goroutineID int) (string, error) {

--- a/service/dap/server_test.go
+++ b/service/dap/server_test.go
@@ -3689,10 +3689,6 @@ func TestPanicBreakpointOnNext(t *testing.T) {
 }
 
 func TestFatalThrowBreakpoint(t *testing.T) {
-	// This is not currently flaky for Go 1.17 see https://github.com/golang/go/issues/46425.
-	if goversion.VersionAfterOrEqual(runtime.Version(), 1, 17) {
-		t.Skip()
-	}
 	runTest(t, "fatalerror", func(client *daptest.Client, fixture protest.Fixture) {
 		runDebugSessionWithBPs(t, client, "launch",
 			// Launch

--- a/service/dap/server_test.go
+++ b/service/dap/server_test.go
@@ -3746,9 +3746,9 @@ func TestFatalThrowBreakpoint(t *testing.T) {
 					client.ContinueRequest(1)
 					client.ExpectContinueResponse(t)
 
-					// TODO(suzmue): Enable this test for 1.17 when https://github.com/golang/go/issues/46425 is fixed.
+					// This does not work for Go 1.16 so skip by detecting versions before or after 1.16.
 					var text string
-					if goversion.VersionAfterOrEqual(runtime.Version(), 1, 17) {
+					if !goversion.VersionAfterOrEqual(runtime.Version(), 1, 16) || goversion.VersionAfterOrEqual(runtime.Version(), 1, 17) {
 						text = "\"all goroutines are asleep - deadlock!\""
 					}
 					se := client.ExpectStoppedEvent(t)

--- a/service/dap/server_test.go
+++ b/service/dap/server_test.go
@@ -3732,9 +3732,6 @@ func TestFatalThrowBreakpoint(t *testing.T) {
 			}})
 	})
 	runTest(t, "testdeadlock", func(client *daptest.Client, fixture protest.Fixture) {
-		if !goversion.VersionAfterOrEqual(runtime.Version(), 1, 17) {
-			t.Skip("broken: see https://github.com/golang/go/issues/46425")
-		}
 		runDebugSessionWithBPs(t, client, "launch",
 			// Launch
 			func() {

--- a/service/dap/server_test.go
+++ b/service/dap/server_test.go
@@ -3736,6 +3736,9 @@ func TestFatalThrowBreakpoint(t *testing.T) {
 			}})
 	})
 	runTest(t, "testdeadlock", func(client *daptest.Client, fixture protest.Fixture) {
+		if !goversion.VersionAfterOrEqual(runtime.Version(), 1, 17) {
+			t.Skip("broken: see https://github.com/golang/go/issues/46425")
+		}
 		runDebugSessionWithBPs(t, client, "launch",
 			// Launch
 			func() {
@@ -3752,7 +3755,7 @@ func TestFatalThrowBreakpoint(t *testing.T) {
 
 					// TODO(suzmue): Enable this test for 1.17 when https://github.com/golang/go/issues/46425 is fixed.
 					var text string
-					if !goversion.VersionAfterOrEqual(runtime.Version(), 1, 16) {
+					if goversion.VersionAfterOrEqual(runtime.Version(), 1, 17) {
 						text = "\"all goroutines are asleep - deadlock!\""
 					}
 					se := client.ExpectStoppedEvent(t)

